src/scripts/fetch/fetch.ts

```diff
@@ -0,0 +1,58 @@
+interface Fetchparams {
+	param: string;
+	value: any;
+}
+
+/**
+ * Fetches data from themoviedb.org API endpoint and returns to the user
+ *
+ * @param URLExtension - the extension over the base API for example: /movie/550
+ * @param APIkey - your api key
+ * @param URLparams array of url parameters to add to fetch request. For example [{param:Query, value:string}]
+ * @returns Returns base object from the API parsed through response.json
+ */
+export const APIFetch = async (
+	URLExtension: string,
+	APIkey: string,
+	URLparams?: Fetchparams[]
+) => {
+	//Basic Fetch URL without params
+	const url = `https://api.themoviedb.org/3${URLExtension}?api_key=${APIkey}`;
+	let finalUrl = "";
+	//reduce additional parameters to string
+
+	if (URLparams !== undefined) {

```

Comment:
could we use !URLparams instead of !==, it will check all other falsy values like null, empty string

Reply:
Yeah good catch!

Reply:
I tried it but no dice

Main issue is that because it's an optional param. Typescript has it typed to `URLParams: Fetchparams[]|undefined` and for some reason doing `(!Urlparams`) doesn't narrow the type down like I need it to.

Reply:
I think the condition here is that Urlparams has some value, so I guess if(Urlparams && Urlparams.length) could also work in favor what you want to achieve. May be you could try once it it works.

Reply:
Worth a look! I'll try it tommorrow!

Reply:
NVM tried it now and it worked! New commit added!